src/software/ai/hl/stp/tactic/shadow_enemy/simulated_shadow_enemy_tactic_test.cpp

```diff
@@ -0,0 +1,192 @@
+                   Timestamp::fromSeconds(0));
+    Robot shadowee(1, Point(0, -2), Vector(0, 0), Angle::fromDegrees(135),
+                   AngularVelocity::zero(), Timestamp::fromSeconds(0));
+    Robot enemy(2, Point(0, 2), Vector(0, 0), Angle::threeQuarter(),
+                AngularVelocity::zero(), Timestamp::fromSeconds(0));
+
+
+    EnemyThreat enemy_threat{shadowee,     true, Angle::zero(), std::nullopt,
+                             std::nullopt, 1,    enemy};
+
+    addFriendlyRobots(
+        {RobotStateWithId{.id = 0, .robot_state = shadower.currentState()}});
+    addEnemyRobots({RobotStateWithId{.id = 1, .robot_state = shadowee.currentState()},
+                    RobotStateWithId{.id = 2, .robot_state = enemy.currentState()}});
+
+
+    setBallState(BallState(Point(0, -2), Vector(0, 0)));
+    auto tactic = std::make_shared<ShadowEnemyTactic>();
+    tactic->updateControlParams(enemy_threat, 2, 0);
+    setTactic(tactic);
+    setRobotId(0);
+
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
+        [this, tactic, shadowee](std::shared_ptr<World> world_ptr,
+                                 ValidationCoroutine::push_type& yield) {
+            // As the shadowee is located at (0,-2), we first find the shot
+            // vector to our net and then normalize this vector to a distance
+            // of 2 away from the shadowee
+            Vector shot       = field().friendlyGoalCenter() - shadowee.position();
+            Point destination = shadowee.position() + shot.normalize(2);
+            robotAtPosition(0, world_ptr, destination, 0.01, yield);
+        }};
+
+    std::vector<ValidationFunction> non_terminating_validation_functions = {};
+
+    runTest(terminating_validation_functions, non_terminating_validation_functions,
+            Duration::fromSeconds(5));
+}
+
+
+TEST_F(SimulatedShadowEnemyTacticTest, test_steal_and_chip)
+{
+    Robot shadower(0, Point(-2, 0), Vector(0, 0), Angle::zero(), AngularVelocity::zero(),
+                   Timestamp::fromSeconds(0));
+    Robot shadowee(1, Point(0, -2), Vector(0, 0), Angle::quarter(),
+                   AngularVelocity::zero(), Timestamp::fromSeconds(0));
+    Robot enemy(2, Point(0, 2), Vector(0, 0), Angle::threeQuarter(),
+                AngularVelocity::zero(), Timestamp::fromSeconds(0));
+
+
+    EnemyThreat enemy_threat{shadowee,     true, Angle::zero(), std::nullopt,
+                             std::nullopt, 1,    enemy};
+
+    addFriendlyRobots(
+        {RobotStateWithId{.id = 0, .robot_state = shadower.currentState()}});
+    addEnemyRobots({RobotStateWithId{.id = 1, .robot_state = shadowee.currentState()},
+                    RobotStateWithId{.id = 2, .robot_state = enemy.currentState()}});
+
+
+    setBallState(BallState(Point(0, -1.75), Vector(0, 0)));
+    auto tactic = std::make_shared<ShadowEnemyTactic>();
+    tactic->updateControlParams(enemy_threat, 2, 100);
```

Comment:
if the steal and chip distance wasn't 100, would it then not steal and chip in this situation?

Reply:
Yes, so the steal_and_chip distance is defined as how close to the enemy before the robot will try to steal and chip the ball, I purposely set it to a large number so that our robot would try to steal and chip

Reply:
In what situations wouldn't we want to steal and chip?

Reply:
If the robot we are shadowing is about to take a shot and the position to block the shot is easier than trying to steal and chip the ball. Also we may decide not to steal and chip the ball since it could potentially leave our goalie exposed to a one touch play

Reply:
can it block the shot and approach the enemy robot at the same time?

Reply:
Yes if we do that I can remove the steal and chip distance. Originally, I added the steal and chip distance as it allows us to tweak how aggressive we want to chase the enemy for the ball. I thought that if we set steal and chip distance to a really large number, it would block the shot first then try to steal and chip, but I think the current behaviour would instantly try to steal and chip without blocking first.

Is trying to blocking the shot then try to steal and chip the ball the intended behaviour? If so, I can make the changes.

Reply:
that makes sense to me @akhilveeraghanta @itsarune @tarakong01 what do you think?

Reply:
Changed, I agree it makes more sense to have it try to steal and chip after it reaches the block shot position